src/index.js

```diff
@@ -12,11 +12,13 @@ makeServer();
 ReactDOM.render(
   <React.StrictMode>
     <Router>
-      <AuthProvider>
-        <DataProvider>
-          <App />
-        </DataProvider>
-      </AuthProvider>
+      <ToastProvider>
+        <AuthProvider>
```

Comment:
suggestion: you can have one common provider component which is wrapper for all the providers

Reply:
Interesting, thank you. 

Code updated.